src/kmedoids_ucb.cpp

```diff
@@ -107,9 +107,6 @@ int KMedoids::getSteps() {
  *  @param loss Loss function to be used e.g. L2
  */
 void KMedoids::setLossFn(std::string loss) {
-  if (loss.at(0) == 'L') {
```

Comment:
https://github.com/ThrunGroup/BanditPAM/blob/f28d3096bf65fcf026d00f2ba3784f995cb13734/src/main.cpp#L58-L62
We read the loss in the `main.cpp`. The condition here checks if the loss starts with `L` following with digits. For example if we enter argument `-l L15`, the condition is satisfied and we assign `loss = 15`. On the other hand if we enter argument which does not satisfy the condition, say, `-l L15p`, then we assign `loss = L15p`. 

Then for the case `loss = 15` we set the `Lp` norm here:
https://github.com/ThrunGroup/BanditPAM/blob/f28d3096bf65fcf026d00f2ba3784f995cb13734/src/kmedoids_ucb.cpp#L116-L118

For the case where `loss = L15p`, we may assume this loss is not recognized and handled by the following
https://github.com/ThrunGroup/BanditPAM/blob/9d1e6c86a915ffbee877f66f447ecd917c1aeb96/src/kmedoids_ucb.cpp#L125-L127

Here are some examples I have tried:
```
1. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L2: Medoids: 694,800,306,714,324,959,737,527,168,251
2. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L15: Medoids: 194,319,558,60,133,896,448,475,251,354
3. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l manhattan: Medoids: 737,246,527,168,714,306,838,324,959,800
4. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l cos: Medoids: 567,182,839,340,385,979,915,758,609,649
5. ./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L15p: libc++abi.dylib: terminating with uncaught exception of type char const*
```

The last one does not throw the expected message because there is no `std::invalid_argument()` to handle the message and `catch()` for the exception, I will push a new code to fix it.

Reply:
Ah, great! The second code block was what I missed. Could you also delete the comments right after line 118 in kmedoids_ucb.cpp?

New error message sounds good as well

Reply:
Cool! Then maybe I just keep that error message and remove the comments right after line 118.

Reply:
Actually I think a better error message would still help users

Reply:
Agree. I have edited the code to throw the error messages. Now we will have proper error message for case like `loss = L15p`:
```
./BanditPAM -f ../../data/MNIST-1k.csv -k 10 -v 1 -l L15p:  error: unrecognized loss function
```

Also, I have merged with the updated main branch.